mcc/odin/odin/snapshot_list.go

```diff
@@ -0,0 +1,18 @@
+package odin
+
+import (
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/aws/aws-sdk-go/service/rds/rdsiface"
+)
+
+// ListSnapshots return a list of all DBSnapshots.
+func ListSnapshots(
+	instanceName string,
+	svc rdsiface.RDSAPI,
+) (
+	result []*rds.DBSnapshot,
+	err error,
+) {
+	result = make([]*rds.DBSnapshot, 0)
```

Comment:
`make` keyword reserves a slice of memory and is required for maps and arrays if you want to be able to resize them afterwards. Another benefit is that you can hint how much space are you going to need to prevent extra memory allocations once the array/map grows.

If you don't want to resize them you need to declare them with all their contents at compile time.